presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java

```diff
@@ -154,8 +158,9 @@ public void checkCanShowSchemas(ConnectorSecurityContext context)
     @Override
     public void checkCanShowCreateTable(ConnectorSecurityContext context, SchemaTableName tableName)
     {
+        boolean grantOptionRequired = hiveConfig.getGrantOption();
         // This should really be OWNERSHIP, but Hive uses `SELECT with GRANT`
-        if (!checkTablePermission(context, tableName, SELECT, true)) {
+        if (!checkTablePermission(context, tableName, SELECT, grantOptionRequired)) {
```

Comment:
This is unexpected. Why do we need grant option for `show crate table`. It seems we should just have `false` here.

Reply:
I think this is how it is implemented in Hive. In general Presto requires ownership privilege for `SHOW CREATE...` statement.

Reply:
Indeed